pkg/operator/controller/ingress/deployment.go

```diff
@@ -51,7 +51,11 @@ func (r *reconciler) ensureRouterDeployment(ci *operatorv1.IngressController, in
 	if err != nil {
 		return false, nil, err
 	}
-	desired, err := desiredRouterDeployment(ci, r.Config.IngressControllerImage, infraConfig, ingressConfig, apiConfig, networkConfig)
+	needed, err := r.IsProxyProtocolNeeded(ci, infraConfig)
```

Comment:
It might be simpler to do `platformStatus := oputil.GetPlatformStatus(r.client, infraConfig)` here and then pass `platformStatus` to `IsProxyProtocolNeeded` and to `desiredRouterDeployment`.  It's fine if you prefer your current approach though.

Reply:
@Miciah thanks for the review. Commit `ef1afa0` addresses this feedbck.